src/core/Segments.tsx

```diff
@@ -0,0 +1,103 @@
+import * as THREE from 'three'
+import * as React from 'react'
+import { extend, useFrame } from '@react-three/fiber'
+import { Line2, LineSegmentsGeometry, LineMaterial, LineMaterialParameters } from 'three-stdlib'
+
+type SegmentsProps = {
+  limit?: number
+  lineWidth?: number
+  children: React.ReactNode
+}
+
+type Api = {
+  subscribe: (ref) => void
+}
+
+type Segment = {
+  start: THREE.Vector3 | [number, number, number]
+  end: THREE.Vector3 | [number, number, number]
+  color?: THREE.Color | [number, number, number]
+}
+
+type SegmentRef = React.RefObject<Segment>
+type SegmentProps = Segment
+
+const context = React.createContext<Api>(null!)
+
+const arrColor = (color) => (color instanceof THREE.Color ? color.toArray() : color)
+const arrPos = (pos) => (pos instanceof THREE.Vector3 ? pos.toArray() : pos)
+
+const Segments = React.forwardRef<Line2, SegmentsProps>((props, forwardedRef) => {
+  const { limit = 1000, lineWidth = 1.0, children, ...rest } = props
+  const [segments, setSegments] = React.useState<Array<SegmentRef>>([])
+
+  const [line] = React.useState(() => new Line2())
+  const [material] = React.useState(() => new LineMaterial())
+  const [geometry] = React.useState(() => new LineSegmentsGeometry())
+  const [resolution] = React.useState(() => new THREE.Vector2(512, 512))
+  const [positions] = React.useState(() => Array(limit).fill(0))
+  const [colors] = React.useState(() => Array(limit).fill(1))
+
+  const api = React.useMemo(
+    () => ({
+      subscribe: (ref) => {
+        setSegments((segments) => [...segments, ref])
+        return () => setSegments((segments) => segments.filter((item) => item.current !== ref.current))
+      },
+    }),
+    []
+  )
+
+  useFrame(() => {
+    positions.length = segments.length * 6
+    colors.length = segments.length * 6
+    segments.forEach((segmentRef, i) => {
+      const segment = segmentRef.current
+      if (segment) {
+        const segmentStart = arrPos(segment.start)
+        const segmentEnd = arrPos(segment.end)
+        const segmentColor = arrColor(segment.color)
```

Comment:
This doesn't work when passing a string color

Reply:
It currently works the same way as vertexColors in the Line component. But yes it would be nice to pass strings as well.